modules/network-messages/src/integrationTest/resources/org/apache/ignite/network/messages/internal/processor/AllTypesMessage.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.network.messages.internal.processor;
+
+import java.util.BitSet;
+import java.util.Collection;
+import java.util.Map;
+import java.util.UUID;
+import org.apache.ignite.lang.IgniteUuid;
+import org.apache.ignite.network.messages.NetworkMessage;
+import org.apache.ignite.network.messages.annotations.AutoSerializable;
+
+@AutoSerializable(messageFactory = AllTypesMessageFactory.class)
+public interface AllTypesMessage extends NetworkMessage {
+    short TYPE = 123;
+
+    byte a();
+
+    short b();
+
+    int c();
+
+    long d();
+
+    float e();
+
+    double f();
+
+    char g();
+
+    boolean h();
+
+    byte[] i();
+
+    short[] j();
+
+    int[] k();
+
+    long[] l();
+
+    float[] m();
+
+    double[] n();
+
+    char[] o();
+
+    boolean[] p();
+
+    String q();
+
+    BitSet r();
+
+    UUID s();
+
+    IgniteUuid t();
+
+    NetworkMessage u();
+
+    NetworkMessage[] v();
+
+    Collection<NetworkMessage> w();
+
+    Map<String, NetworkMessage> x();
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override public default short directType() {
+        return 5555;
+    }
+
+    interface Builder {
```

Comment:
Can you elaborate, please?

Reply:
Sure, I think Builder declaration inside interface is redundant. Declaring it inside Impl class should be enough. I doubt there will be more than one implementation for one message interface. 
So instead of: 
```
Message.Builder builder = new MessageImpl();
```
the code will look like this:
```
MessageImpl.Builder builder = new MessageImpl();
```

Reply:
ok, I'll think about this in the next ticket. Here I'm just mimicking the approach used in the raf module...

Reply:
But how are you going to use the setters, if `MessageImpl` will be a private class? Also, I think that Builder will become a separate class, when we are going to generate the implementations, because current approach with extending Builders makes little sense (why not use setters instead?)

Reply:
I agree with Alexander, right now this is the best that we can do